src/pdftoraw.c

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (C) 2021 Thierry HUCHARD <thierry@ordissimo.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <unistd.h>
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <getopt.h>
+#include <gio/gio.h>
+#include <poppler.h>
+#include <gdk/gdk.h>
+
+
+// 1 inch = 72 points
+#define PTS 72.0
+
+// Resolution for the printer, 150 my choice
+#define DPI 150.0
+
+static gchar *filename = "";
+static gboolean opt_test = FALSE;
+static gboolean opt_pages = FALSE;
+static int opt_raw = -1;
+static int opt_width = -1;
+static int opt_height = -1;
+
+static GOptionEntry entries[] = {
+  { "file", 'f', 0, G_OPTION_ARG_STRING, &filename, "PDF file path", NULL },
+  { "test", 't', 0, G_OPTION_ARG_NONE, &opt_test, "Test if the file is a PDF file", NULL },
+  { "pages", 'p', 0, G_OPTION_ARG_NONE, &opt_pages, "Returns the number of pages in the PDF file", NULL },
+  { "width", 'w', 0, G_OPTION_ARG_INT, &opt_width, "Give the size of the page provided in arguments", NULL},
+  { "height", 'W', 0, G_OPTION_ARG_INT, &opt_height, "Gives the height of the page provided in arguments", NULL},
+  { "raw", 'r', 0, G_OPTION_ARG_INT, &opt_raw, "Retrieves data in pixels from the page provided as arguments", NULL},
+  { NULL }
+};
+
+static void
+print_usage (void)
```

Comment:
GOptionContext already prints its own help, you don't need to do it.

Reply:
You can use `g_option_context_get_help()` if you want it as a string to print.

Reply:
Fixed by 9cb421f32dd6f7041dedf210089e6fe1ca5bb27b